src/pex-builder/builder/source.py

```diff
@@ -12,6 +12,9 @@
 
 from . import util
 
+# inspired by https://github.com/github/gitignore/blob/main/Python.gitignore
+IGNORED_PATTERNS = [".git", "__pycache__", ".pytest_cache", ".tox", ".nox", "*.pyc", ".mypy_cache"]
```

Comment:
I don't know what spec the pattern matching needs to conform to (regex? glob?) but it'd be slick if we read the actual `.gitignore` and respected its patterns. Instead of maintaining our own list of commonly ignored directories.

We can optimize for something like that later though.